cangw.c

```diff
@@ -264,7 +264,7 @@ void print_usage(char *prg)
 	fprintf(stderr, "  <instruction>  is one of 'AND' 'OR' 'XOR' 'SET'\n");
 	fprintf(stderr, "  <canfd_frame-elements>  is _one_ or _more_ of 'I'd 'F'lags 'L'ength 'D'ata\n");
 	fprintf(stderr, "  <can_id>  is an u32 value containing the CAN FD Identifier\n");
-	fprintf(stderr, "  <flags>  is an u8 value containing CAN FD flags (CANFD_BRS, CANFD_ESI)\n");
+	fprintf(stderr, "  <flags>  is an u8 value containing CAN FD flags (CANFD_BRS, CANFD_ESI, CANFD_FDF)\n");
```

Comment:
This only makes sense when the flag has landed in the mainline kernel.
Btw. I'm not sure if this bit should have a meaning inside the Linux kernel anyway.
The intention is a usage to help user space programs to handle CAN/CANFD frames inside struct canfd_frames.

Reply:
We might set the CANFD_FDF flag inside the alloc_canfd_skb() function....

https://github.com/torvalds/linux/blob/master/drivers/net/can/dev/skb.c#L210

Reply:
Yes. But we then introduce a flag which has not been set since CAN FD has been supported in Linux 3.6.
Don't get me wrong: I'm not against setting this flag. But we might run into a problem, when user-space programs *rely* on this bit. Therefore my use-case was to 'define and reserve' this bit for user space programs that like to use the canfd_frame structure for dual-use (CAN/CANFD). These programs then know, that they can rely on this bit - independent from the kernel version.

Reply:
Btw. we might use that bit for kernel internal use cases and improvements too. Why not?
But it has to be clear that user space can not rely on this.

Reply:
Would you consider to go further with that patch from 2017?
https://lore.kernel.org/linux-can/20170411134343.3089-1-socketcan@hartkopp.net/ 
At least it still applies with an offset of 11 lines ;-)